pkg/git/repository.go

```diff
@@ -136,16 +140,30 @@ func (o *CloneOptions) Clone(ctx context.Context) (Repository, fs.FS, error) {
 
 	r, err := clone(ctx, o)
 	if err != nil {
-		if err == transport.ErrEmptyRemoteRepository {
-			log.G(ctx).Debug("empty repository, initializing new one with specified remote")
+		switch err {
+		case transport.ErrRepositoryNotFound:
+			if !o.AutoCreate {
+				return nil, nil, err
+			}
+
+			log.G(ctx).Debug("no repository, creating a new one")
+			_, err = createRepo(ctx, o)
+			if err != nil {
```

Comment:
the error here is `transport.ErrEmptyRemoteRepository`?

Reply:
No, creating a repo does not return `transport.ErrEmptyRemoteRepository`
It should return the cloneURL and no error. If it returns an error, it means the creation failed.

Reply:
so what does the fallthrough help with?

Reply:
nvm: https://yourbasic.org/golang/switch-statement/#fallthrough